File: LMSApi/LMSApi.ModelLibrary/Models/Assignments/Assignment.cs
using LMSApi.ModelLibrary.Enums;

namespace LMSApi.ModelLibrary.Models
{
    public class Assignments
    {
        public int Id { get; set; }
        public int CourseSectionId { get; set; }
        public string Title { get; set; }
        public string? Description { get; set; }
        public string? Instructions { get; set; }
        public bool IsCompulsory { get; set; }
        public int TotalMarks { get; set; }
        public int PassingMarks { get; set; }
        public AssignmentAttachmentType AttachmentType { get; set; } = AssignmentAttachmentType.None;
        public string? AttachmentUrl { get; set; }

        /// <summary>
        /// Number of days from enrollment date within which the student must submit.
        /// 0 means no deadline.
        /// </summary>
        public int DeadlineInDays { get; set; }
        public DateTime? DeadlineDate { get; set; }

        /// <summary>Maximum number of submission attempts allowed per student.</summary>
        public int MaxSubmissions { get; set; } = 1;

        public bool IsLateSubmissionAllowed { get; set; }
        public int SortOrder { get; set; }
        public PublishStatus Status { get; set; } = PublishStatus.Draft;

        public DateTime CreatedAt { get; set; }

        // Navigation properties
        public CourseSection CourseSection { get; set; }
        public ICollection<AssignmentSubmissions> Submissions { get; set; } = new List<AssignmentSubmissions>();
    }
}

File: lms-app/src/app/models/assignment.ts
import { AssignmentAttachmentType } from '../enums/assignment-attachment-type.enum';

export interface AssignmentResponse {
  id: number;
  courseSectionId: number;
  title: string;
  description?: string;
  instructions?: string;
  isCompulsory: boolean;
  totalMarks: number;
  passingMarks: number;
  attachmentType: AssignmentAttachmentType | number; // AssignmentAttachmentType
  attachmentUrl?: string;
  deadlineInDays: number;
  deadlineDate?: string;
  maxSubmissions: number;
  isLateSubmissionAllowed: boolean;
  status: number;
  createdAt: string;
}

export interface AssignmentSubmissionResponse {
  id: number;
  assignmentId: number;
  studentId: number;
  submissionText?: string;
  attachmentType?: AssignmentAttachmentType | number; // AssignmentAttachmentType
  submittedAssignmentUrl?: string;
  marksAwarded?: number;
  feedback?: string;
  submittedAt: string;
  gradedAt?: string;
  status: string; // "Submitted", "UnderReview", "Graded", etc.
  isPassed?: boolean;
  attemptNumber: number;
  studentName?: string;
  studentEmail?: string;
  isLate?: boolean;
  studentDeadline?: string;
  courseTitle?: string;
  sectionTitle?: string;
  assignmentTitle?: string;
}

export interface AssignmentStatusResponse {
  assignmentId: number;
  studentId: number;
  attemptsMade: number;
  maxSubmissions: number;
  remainingAttempts: number;
  isPassed?: boolean;
  latestStatus?: string;
  deadline?: string;
}

export interface InstructorAssignmentSummaryDto {
  id: number;
  courseSectionId: number;
  title: string;
  courseTitle: string;
  sectionTitle: string;
  totalMarks: number;
  deadlineInDays: number;
  deadlineDate?: string;
  pendingSubmissionsCount: number;
  status: number;
  createdAt: string;
}

export interface PagedInstructorAssignmentResponse {
  assignments: InstructorAssignmentSummaryDto[];
  totalCount: number;
  pageNumber: number;
  pageSize: number;
  totalPages: number;
  totalPendingCount: number;
  fullyGradedCount: number;
  uniqueCourseCount: number;
}
